alacritty_terminal/src/keyboard_motion.rs

```diff
@@ -0,0 +1,767 @@
+use std::cmp::min;
+
+use serde::Deserialize;
+
+use crate::event::EventListener;
+use crate::grid::{GridCell, Scroll};
+use crate::index::{Column, Point};
+use crate::term::cell::Flags;
+use crate::term::{Search, Term};
+
+/// Possible keyboard motion movements.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Deserialize)]
+pub enum KeyboardMotion {
+    /// Move up.
+    Up,
+    /// Move down.
+    Down,
+    /// Move left.
+    Left,
+    /// Move right.
+    Right,
+    /// Move to start of line.
+    First,
+    /// Move to end of line.
+    Last,
```

Comment:
Why not call these `Home` and `End`? These names are much more standard.

Reply:
Those are key names, not positions. Neither of those are bound by default in Alacritty.

That just seems confusing.

Reply:
`LastOccupied` is also much, much nicer than `EndOccupied`, which is incredibly confusing.

Reply:
There isn't currently `LastOccupied`, only `FirstOccupied`, but point taken. We should probably add  `FirstOccupied` right?

Following `vi`'s keybindings, `FirstOccupied` should be bound to <kbd>^</kbd>. Unfortunatly, the much more useful (I think) `LastOccupied`'s default binding in `vi` isn't compatible with our bindings since it's <kbd>g</kbd> followed by <kbd>_</kbd>.